java-client/src/main/java/co/elastic/clients/transport/TransportOptions.java

```diff
@@ -38,6 +38,8 @@ public interface TransportOptions {
 
     Function<List<String>, Boolean> onWarnings();
 
+    boolean keepResponseBodyOnException();
```

Comment:
This option isn't straightforward and deserves some javadocs

```suggestion
    /**
     * If {@code true}, the response body in {@code TransportException.response().body()} is guaranteed to be
     * replayable (i.e. buffered), even if the original response was streamed. This allows inspecting the
     * response body in case of error.
     */
    boolean keepResponseBodyOnException();
```